### DISCUSSION TITLE
Dealing with breaking changes and version numbers

### DISCUSSION THREAD

**Discussion by lefou** [build-server-protocol/build-server-protocol#608] [General]
Dear BSP community,

the latest bsp4j release 2.1.0-M6 contains breaking changes. I really really hope, this is due to the milestone character and the leading version will change to a `3.0.0` once it get stable. 

I think, as tooling developers, we should know the best how problematic and frustrating braking changes in minor releases can be. Hence, I politely urge you to decouple the versions of the support libraries from the BSP version. If you fear that there might be irritation for the library users, you could still keep the protocol version in the artifact name, but please don't publish breaking changes in minor releases.

Regards,
Tobias

**Comment by jastice** [comment]
For reference, PR and discussion around it: https://github.com/build-server-protocol/build-server-protocol/pull/555

**Comment by jastice** [comment]
Agree we should decouple protocol and lib versions.

IIRC we added a protocol version number to the library, which should help disambiguate.
Is there a reasonable way to add the protocol version to the artifact name? Would it cause any issues?

**Comment by agluszak** [comment]
I drafted a new release https://github.com/build-server-protocol/build-server-protocol/releases/edit/untagged-663cb6144f3802d4570f which simply promoted v2.1.0-M5 (the last before smithy) to v2.1.0

**Reply by adpi2** [reply]
I allowed myself to click the `generate release notes` button to generate the list of PRs and new contributors. So the link to the draft release is now: https://github.com/build-server-protocol/build-server-protocol/releases/tag/untagged-061d443c9c7518d0178a

**Reply by lefou** [reply]
I can't see the draft-release, but I like the stable release anyways.

**Reply by agluszak** [reply]
@lefou probably that's because you haven't been added to the contributors group in this repo. I'll check if I can add you, but probably an admin has to do it. @jastice ?
